In [1]:
import pandas as pd
import numpy as np

In [2]:
df_psy = pd.read_csv('Youtube01-Psy.csv')

In [3]:
df_psy.head()

COMMENT_ID            AUTHOR  \
0  LZQPQhLyRh80UYxNuaDWhIGQYNQ96IuCg-AYWqNPjpU         Julius NM   
1  LZQPQhLyRh_C2cTtd9MvFRJedxydaVW-2sNg5Diuo4A       adam riyati   
2  LZQPQhLyRh9MSZYnf8djyk0gEF9BHDPYrrK-qCczIY8  Evgeny Murashkin   
3          z13jhp0bxqncu512g22wvzkasxmvvzjaz04   ElNino Melendez   
4          z13fwbwp1oujthgqj04chlngpvzmtt3r3dw            GsMega   

                  DATE                                            CONTENT  \
0  2013-11-07T06:20:48  Huh, anyway check out this you[tube] channel: ...   
1  2013-11-07T12:37:15  Hey guys check out my new channel and our firs...   
2  2013-11-08T17:34:21             just for test I have to say murdev.com   
3  2013-11-09T08:28:43   me shaking my sexy ass on my channel enjoy ^_^    
4  2013-11-10T16:05:38            watch?v=vtaRGgvGtWQ   Check this out .   

   CLASS  
0      1  
1      1  
2      1  
3      1  
4      1

In [4]:
df_psy.shape

(350, 5)

5 columns and 350 rows

In [5]:
df_psy.count()

COMMENT_ID    350
AUTHOR        350
DATE          350
CONTENT       350
CLASS         350
dtype: int64

No Null values present here ;)

In [6]:
df_kp = pd.read_csv('Youtube02-KatyPerry.csv')

In [7]:
df_kp.head()

COMMENT_ID        AUTHOR                 DATE  \
0      z12pgdhovmrktzm3i23es5d5junftft3f   lekanaVEVO1  2014-07-22T15:27:50   
1    z13yx345uxepetggz04ci5rjcxeohzlrtf4      Pyunghee  2014-07-27T01:57:16   
2  z12lsjvi3wa5x1vwh04cibeaqnzrevxajw00k    Erica Ross  2014-07-27T02:51:43   
3    z13jcjuovxbwfr0ge04cev2ipsjdfdurwck  Aviel Haimov  2014-08-01T12:27:48   
4  z13qybua2yfydzxzj04cgfpqdt2syfx53ms0k    John Bello  2014-08-01T21:04:03   

                                             CONTENT  CLASS  
0  i love this so much. AND also I Generate Free ...      1  
1  http://www.billboard.com/articles/columns/pop-...      1  
2  Hey guys! Please join me in my fight to help a...      1  
3  http://psnboss.com/?ref=2tGgp3pV6L this is the...      1  
4  Hey everyone. Watch this trailer!!!!!!!!  http...      1

In [8]:
df_kp.shape

(350, 5)

In [9]:
df_kp.count()

COMMENT_ID    350
AUTHOR        350
DATE          350
CONTENT       350
CLASS         350
dtype: int64

No Null values present here also ;)

In [10]:
df_lmfao = pd.read_csv('Youtube03-LMFAO.csv')

In [11]:
df_lmfao.head()

COMMENT_ID        AUTHOR  \
0  z13uwn2heqndtr5g304ccv5j5kqqzxjadmc0k  Corey Wilson   
1  z124jvczaz3dxhnbc04cffk43oiugj25yzo0k   Epic Gaming   
2      z13tczjy5xj0vjmu5231unho1ofey5zdk     LaS Music   
3  z13tzr0hdpnayhqqc04cd3zqqqjkf3ngckk0k    Cheryl Fox   
4  z12pcvix4zedcjvyb04ccr1r0mr2g5xwyng0k    PATRICK_TW   

                         DATE  \
0  2015-05-28T21:39:52.376000   
1  2015-05-28T20:07:20.610000   
2  2015-05-28T19:23:35.355000   
3  2015-05-28T17:49:35.294000   
4  2015-05-28T16:28:26.818000   

                                             CONTENT  CLASS  
0  <a href="http://www.youtube.com/watch?v=KQ6zr6...      0  
1                                   wierd but funny      0  
2  Hey guys, I&#39;m a human.<br /><br /><br />Bu...      1  
3       Party Rock....lol...who wants to shuffle!!!      0  
4                                        Party rock      0

In [12]:
df_lmfao.shape

(438, 5)

In [13]:
df_lmfao.count()

COMMENT_ID    438
AUTHOR        438
DATE          438
CONTENT       438
CLASS         438
dtype: int64

No Null values present here also ;)

In [14]:
df_em = pd.read_csv('Youtube04-Eminem.csv')

In [15]:
df_em.head()

COMMENT_ID         AUTHOR  \
0    z12rwfnyyrbsefonb232i5ehdxzkjzjs2    Lisa Wellas   
1  z130wpnwwnyuetxcn23xf5k5ynmkdpjrj04   jason graham   
2    z13vsfqirtavjvu0t22ezrgzyorwxhpf3     Ajkal Khan   
3  z12wjzc4eprnvja4304cgbbizuved35wxcs  Dakota Taylor   
4    z13xjfr42z3uxdz2223gx5rrzs3dt5hna    Jihad Naser   

                         DATE  \
0                         NaN   
1  2015-05-29T02:26:10.652000   
2                         NaN   
3  2015-05-29T02:13:07.810000   
4                         NaN   

                                             CONTENT  CLASS  
0          +447935454150 lovely girl talk to me xxx      1  
1    I always end up coming back to this song<br />      0  
2  my sister just received over 6,500 new <a rel=...      1  
3                                              Cool      0  
4                     Hello I&#39;am from Palastine      1

In [16]:
df_em.shape

(448, 5)

In [17]:
df_em.count()

COMMENT_ID    448
AUTHOR        448
DATE          203
CONTENT       448
CLASS         448
dtype: int64

Some missing values are there in the date column

In [18]:
df_sk = pd.read_csv('Youtube05-Shakira.csv')

In [19]:
df_sk.head()

COMMENT_ID                              AUTHOR  \
0    z13lgffb5w3ddx1ul22qy1wxspy5cpkz504                          dharma pal   
1      z123dbgb0mqjfxbtz22ucjc5jvzcv3ykj                       Tiza Arellano   
2  z12quxxp2vutflkxv04cihggzt2azl34pms0k  Prìñçeśś Âliś Łøvê Dømíñø Mâđiś™    
3      z12icv3ysqvlwth2c23eddlykyqut5z1h                       Eric Gonzalez   
4      z133stly3kete3tly22petvwdpmghrlli                       Analena López   

                         DATE  \
0  2015-05-29T02:30:18.971000   
1  2015-05-29T00:14:48.748000   
2  2015-05-28T21:00:08.607000   
3  2015-05-28T20:47:12.193000   
4  2015-05-28T17:08:29.827000   

                                             CONTENT  CLASS  
0                                         Nice song      0  
1                                      I love song       0  
2                                      I love song       0  
3  860,000,000 lets make it first female to reach...      0  
4                      shakira is best for worldcup      0

In [20]:
df_sk.shape

(370, 5)

In [21]:
df_sk.count()

COMMENT_ID    370
AUTHOR        370
DATE          370
CONTENT       370
CLASS         370
dtype: int64

No missing values ;)

Trying to merge all the datasets into one so that we'll have one large dataset

In [22]:
data_frames = [df_psy, df_kp, df_lmfao, df_em, df_sk]

As we can see all the dataset have the same column names luckily, COMMENT_ID, AUTHOR, DATE, CONTENT, CLASS by this it is easy for us to use the .concat command by pandas to that it will concatenate all the data according to their respective column starting from the "df_psy" up to the "df_sk"

In [23]:
concat_df = pd.concat(data_frames)

In [24]:
concat_df.head()

COMMENT_ID            AUTHOR  \
0  LZQPQhLyRh80UYxNuaDWhIGQYNQ96IuCg-AYWqNPjpU         Julius NM   
1  LZQPQhLyRh_C2cTtd9MvFRJedxydaVW-2sNg5Diuo4A       adam riyati   
2  LZQPQhLyRh9MSZYnf8djyk0gEF9BHDPYrrK-qCczIY8  Evgeny Murashkin   
3          z13jhp0bxqncu512g22wvzkasxmvvzjaz04   ElNino Melendez   
4          z13fwbwp1oujthgqj04chlngpvzmtt3r3dw            GsMega   

                  DATE                                            CONTENT  \
0  2013-11-07T06:20:48  Huh, anyway check out this you[tube] channel: ...   
1  2013-11-07T12:37:15  Hey guys check out my new channel and our firs...   
2  2013-11-08T17:34:21             just for test I have to say murdev.com   
3  2013-11-09T08:28:43   me shaking my sexy ass on my channel enjoy ^_^    
4  2013-11-10T16:05:38            watch?v=vtaRGgvGtWQ   Check this out .   

   CLASS  
0      1  
1      1  
2      1  
3      1  
4      1

In [25]:
concat_df.shape

(1956, 5)

In [26]:
concat_df.count()

COMMENT_ID    1956
AUTHOR        1956
DATE          1711
CONTENT       1956
CLASS         1956
dtype: int64

Look we've now a total of 1956 rows, that is sum up from the separately of 5 datasets.
The DATE column was having some missing values in the 'Youtube04-Eminem' dataset.

In [27]:
concat_df

COMMENT_ID                AUTHOR  \
0    LZQPQhLyRh80UYxNuaDWhIGQYNQ96IuCg-AYWqNPjpU             Julius NM   
1    LZQPQhLyRh_C2cTtd9MvFRJedxydaVW-2sNg5Diuo4A           adam riyati   
2    LZQPQhLyRh9MSZYnf8djyk0gEF9BHDPYrrK-qCczIY8      Evgeny Murashkin   
3            z13jhp0bxqncu512g22wvzkasxmvvzjaz04       ElNino Melendez   
4            z13fwbwp1oujthgqj04chlngpvzmtt3r3dw                GsMega   
..                                           ...                   ...   
365  _2viQ_Qnc6-bMSjqyL1NKj57ROicCSJV5SwTrw-RFFA          Katie Mettam   
366  _2viQ_Qnc6-pY-1yR6K2FhmC5i48-WuNx5CumlHLDAI  Sabina Pearson-Smith   
367  _2viQ_Qnc6_k_n_Bse9zVhJP8tJReZpo8uM2uZfnzDs         jeffrey jules   
368  _2viQ_Qnc6_yBt8UGMWyg3vh0PulTqcqyQtdE7d4Fl0        Aishlin Maciel   
369  _2viQ_Qnc685RPw1aSa1tfrIuHXRvAQ2rPT9R06KTqA           Latin Bosch   

                           DATE  \
0           2013-11-07T06:20:48   
1           2013-11-07T12:37:15   
2           2013-11-08T17:34:21   
3           2013-11-09T08:28:43   
4           2013-11-10T16:05:38   
..                          ...   
365  2013-07-13T13:27:39.441000   
366  2013-07-13T13:14:30.021000   
367  2013-07-13T12:09:31.188000   
368  2013-07-13T11:17:52.308000   
369  2013-07-12T22:33:27.916000   

                                               CONTENT  CLASS  
0    Huh, anyway check out this you[tube] channel: ...      1  
1    Hey guys check out my new channel and our firs...      1  
2               just for test I have to say murdev.com      1  
3     me shaking my sexy ass on my channel enjoy ^_^       1  
4              watch?v=vtaRGgvGtWQ   Check this out .      1  
..                                                 ...    ...  
365  I love this song because we sing it at Camp al...      0  
366  I love this song for two reasons: 1.it is abou...      0  
367                                                wow      0  
368                            Shakira u are so wiredo      0  
369                         Shakira is the best dancer      0  

[1956 rows x 5 columns]

Assigning keys to every datasets' comments in the concatenated dataset so that if we need to see the comments of a specific person or say dataset it would be easy for us to do

In [28]:
keys = ['psy', 'kattyPerry', 'lmfao', 'eminem', 'shakira']

In [29]:
# by default the keys parameter is set to none
final_dataframe = pd.concat(data_frames, keys=keys)

In [30]:
final_dataframe.count()

COMMENT_ID    1956
AUTHOR        1956
DATE          1711
CONTENT       1956
CLASS         1956
dtype: int64

Nothing is changed with the dataset's columns and rows

But now we can see the comments of Katty Perry from the final large dataset

In [31]:
final_dataframe.loc['kattyPerry']

COMMENT_ID               AUTHOR  \
0        z12pgdhovmrktzm3i23es5d5junftft3f          lekanaVEVO1   
1      z13yx345uxepetggz04ci5rjcxeohzlrtf4             Pyunghee   
2    z12lsjvi3wa5x1vwh04cibeaqnzrevxajw00k           Erica Ross   
3      z13jcjuovxbwfr0ge04cev2ipsjdfdurwck         Aviel Haimov   
4    z13qybua2yfydzxzj04cgfpqdt2syfx53ms0k           John Bello   
..                                     ...                  ...   
345    z12sjp3zgtqnvlysj23zuxxaolrvd1oj504          Kacy Cluley   
346  z132enrpoy35yxpoe04cjr4zur3jvbyq3xo0k    Kasia Fabisiewicz   
347      z132jbmxfqm4fjysg23nwjfb2mv2vxnua  Decio Alves Martins   
348    z12cdlswetvnejcri04cex0jfwy2u3tzj54         Rafi Hossain   
349  z120e5uautvcuper304ccf4bjrjugdpbwrc0k           moaz adnan   

                    DATE                                            CONTENT  \
0    2014-07-22T15:27:50  i love this so much. AND also I Generate Free ...   
1    2014-07-27T01:57:16  http://www.billboard.com/articles/columns/pop-...   
2    2014-07-27T02:51:43  Hey guys! Please join me in my fight to help a...   
3    2014-08-01T12:27:48  http://psnboss.com/?ref=2tGgp3pV6L this is the...   
4    2014-08-01T21:04:03  Hey everyone. Watch this trailer!!!!!!!!  http...   
..                   ...                                                ...   
345  2015-06-05T18:59:52  This song means so much to me thank you  soooo...   
346  2015-06-05T19:02:05                                             &lt;3   
347  2015-06-05T19:29:20  KATY PERRY, I AM THE "DÉCIO CABELO", "DECIO HA...   
348  2015-06-05T19:55:08  Honestly speaking except taylor swift and adel...   
349  2015-06-05T20:01:23  who is going to reach the billion first : katy...   

     CLASS  
0        1  
1        1  
2        1  
3        1  
4        1  
..     ...  
345      0  
346      0  
347      1  
348      0  
349      0  

[350 rows x 5 columns]

In [32]:
# Look in above code it print out the comments ralted to the key==kattyPerry. 
# And for confirmation I run this line of code
df_kp.head()

COMMENT_ID        AUTHOR                 DATE  \
0      z12pgdhovmrktzm3i23es5d5junftft3f   lekanaVEVO1  2014-07-22T15:27:50   
1    z13yx345uxepetggz04ci5rjcxeohzlrtf4      Pyunghee  2014-07-27T01:57:16   
2  z12lsjvi3wa5x1vwh04cibeaqnzrevxajw00k    Erica Ross  2014-07-27T02:51:43   
3    z13jcjuovxbwfr0ge04cev2ipsjdfdurwck  Aviel Haimov  2014-08-01T12:27:48   
4  z13qybua2yfydzxzj04cgfpqdt2syfx53ms0k    John Bello  2014-08-01T21:04:03   

                                             CONTENT  CLASS  
0  i love this so much. AND also I Generate Free ...      1  
1  http://www.billboard.com/articles/columns/pop-...      1  
2  Hey guys! Please join me in my fight to help a...      1  
3  http://psnboss.com/?ref=2tGgp3pV6L this is the...      1  
4  Hey everyone. Watch this trailer!!!!!!!!  http...      1

Now save the final_dataframe into a csv file by the pandas method to_csv

In [33]:
final_dataframe.to_csv('final_dataframe.csv')

In [34]:
final_dataframe.size

9780

## Cleaning the data

In [35]:
final_dataframe.columns

Index(['COMMENT_ID', 'AUTHOR', 'DATE', 'CONTENT', 'CLASS'], dtype='object')

In [36]:
final_dataframe.dtypes

COMMENT_ID    object
AUTHOR        object
DATE          object
CONTENT       object
CLASS          int64
dtype: object

In [37]:
final_dataframe.isnull().isnull().sum()

COMMENT_ID    0
AUTHOR        0
DATE          0
CONTENT       0
CLASS         0
dtype: int64

In [38]:
final_dataframe.isnull().sum()

COMMENT_ID      0
AUTHOR          0
DATE          245
CONTENT         0
CLASS           0
dtype: int64

In [39]:
final_dataframe.DATE

psy      0             2013-11-07T06:20:48
         1             2013-11-07T12:37:15
         2             2013-11-08T17:34:21
         3             2013-11-09T08:28:43
         4             2013-11-10T16:05:38
                           ...            
shakira  365    2013-07-13T13:27:39.441000
         366    2013-07-13T13:14:30.021000
         367    2013-07-13T12:09:31.188000
         368    2013-07-13T11:17:52.308000
         369    2013-07-12T22:33:27.916000
Name: DATE, Length: 1956, dtype: object

In [40]:
final_dataframe.AUTHOR

psy      0                 Julius NM
         1               adam riyati
         2          Evgeny Murashkin
         3           ElNino Melendez
         4                    GsMega
                        ...         
shakira  365            Katie Mettam
         366    Sabina Pearson-Smith
         367           jeffrey jules
         368          Aishlin Maciel
         369             Latin Bosch
Name: AUTHOR, Length: 1956, dtype: object

### Contents


As if look up the we can observe that from the contents, we'll classiy the comments as spam or not spam. 
Like the other attributes doesn't matter that much. As the one is name, we don't care about who is the author. Also in case of 
date it also doesn't matter like on which date it has been posted or something else.
So for us the 'CONTENT' and the 'CLASS' columns are most important.


In [41]:
data_df = final_dataframe[['CONTENT', 'CLASS']]

In [42]:
data_df.head()

CONTENT  CLASS
psy 0  Huh, anyway check out this you[tube] channel: ...      1
    1  Hey guys check out my new channel and our firs...      1
    2             just for test I have to say murdev.com      1
    3   me shaking my sexy ass on my channel enjoy ^_^       1
    4            watch?v=vtaRGgvGtWQ   Check this out .      1

In [43]:
data_df.count()

CONTENT    1956
CLASS      1956
dtype: int64

In [44]:
data_df.columns

Index(['CONTENT', 'CLASS'], dtype='object')

In [45]:
data_feature = data_df['CONTENT']
data_label = data_df['CLASS']

### Extracting features from text

In [46]:
# ML Packages For Vectorization of Text For Feature Extraction
from sklearn.feature_extraction.text import CountVectorizer
from sklearn.feature_extraction.text import TfidfVectorizer

Extracting the features with the CountVectorizer()

In [47]:
corpus = data_feature

In [48]:
vectorizer = CountVectorizer()

In [49]:
# fit the data (feature) which is in corpus 
X = vectorizer.fit_transform(corpus)

In [50]:
# Here are the features it extracted from the data
vectorizer.get_feature_names()

['00',
 '000',
 '002',
 '018',
 '02',
 '034',
 '04',
 '047000',
 '05',
 '053012',
 '0687119038',
 '08',
 '09',
 '0cb8qfjaa',
 '0d878a889c',
 '0dbhjzdw0lbsjbi40gxm0d0p5krhv8xinqli53__wqbahs8zx4mjhw5vwrkpxfoeks',
 '0laviqu2b',
 '10',
 '100',
 '1000',
 '10000000',
 '1000000000',
 '100000415527985',
 '100005244783212',
 '100007085325116',
 '10001',
 '100877300245414',
 '101721377578919894134',
 '10200253113705769',
 '1030',
 '104999962146104962510',
 '10626048',
 '10626835',
 '106865403',
 '107297364',
 '1073741825',
 '1073741828',
 '1073741830',
 '1073741943',
 '108k',
 '109',
 '10b35481',
 '11',
 '1111',
 '1111111111111111111',
 '111719098841907',
 '111982027348137311818',
 '112720997191206369631',
 '11cpwb',
 '11th',
 '12',
 '123',
 '124',
 '124923004',
 '126',
 '127',
 '128gb',
 '12year',
 '13',
 '13017194',
 '131275322914',
 '131338190916',
 '1337',
 '1340488',
 '1340489',
 '1340490',
 '1340491',
 '1340492',
 '1340493',
 '1340494',
 '1340499',
 '1340500',
 '1340502',
 '1340503',
 '134

In [51]:
X.toarray()

array([[0, 0, 0, ..., 0, 0, 0],
       [0, 0, 0, ..., 0, 0, 0],
       [0, 0, 0, ..., 0, 0, 0],
       ...,
       [0, 0, 0, ..., 0, 0, 0],
       [0, 0, 0, ..., 0, 0, 0],
       [0, 0, 0, ..., 0, 0, 0]], dtype=int64)

Now we extracted features from the data. And as we know the models perform well over the numbers and guess what, we've now 
numbers in the form of array of arrays. And so these numbers i-e 'X' will be act as features while the corresponding labels are stored 
in the variable 'data_label'

### Building the Model

In [52]:
# for splitting the data into training and testing part
from sklearn.model_selection import train_test_split

In [53]:
X_train, X_test, y_train, y_test = train_test_split(X, data_label, test_size=0.33, random_state=42)

In [54]:
X_train

<1310x4454 sparse matrix of type '<class 'numpy.int64'>'
	with 17525 stored elements in Compressed Sparse Row format>

In [55]:
X_train.shape

(1310, 4454)

Look we've now 4454 of total features to predict the spam or not spam class of a comment

In [56]:
X_train

<1310x4454 sparse matrix of type '<class 'numpy.int64'>'
	with 17525 stored elements in Compressed Sparse Row format>

The Standard Naive Bayes Classifier for the Spam filtering

In [57]:
from sklearn.naive_bayes import MultinomialNB

In [58]:
clf = MultinomialNB()

In [59]:
clf.fit(X_train, y_train)

MultinomialNB()

In [60]:
accuracy = clf.score(X_test, y_test)

In [61]:
print(accuracy*100, '%')

91.95046439628483 %


#### Prediction

In [62]:
clf.predict(X_test)

array([0, 0, 0, 1, 1, 0, 1, 1, 0, 1, 1, 0, 1, 0, 0, 1, 1, 0, 1, 0, 1, 0,
       1, 1, 1, 0, 0, 0, 0, 1, 1, 1, 1, 0, 1, 1, 1, 1, 0, 0, 0, 1, 0, 1,
       1, 1, 1, 1, 0, 1, 1, 0, 1, 1, 0, 0, 1, 0, 1, 1, 1, 1, 1, 1, 0, 0,
       0, 1, 0, 1, 0, 1, 1, 0, 1, 0, 1, 0, 0, 0, 0, 0, 1, 0, 1, 1, 1, 1,
       1, 1, 0, 1, 1, 0, 0, 0, 1, 0, 0, 1, 1, 1, 1, 0, 0, 0, 1, 1, 1, 1,
       0, 1, 0, 1, 0, 0, 0, 1, 0, 1, 1, 1, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1,
       1, 0, 1, 0, 1, 1, 0, 1, 1, 1, 1, 0, 1, 0, 1, 0, 1, 1, 0, 0, 0, 1,
       1, 1, 1, 1, 1, 0, 1, 1, 0, 1, 1, 1, 0, 1, 1, 1, 0, 0, 0, 1, 0, 1,
       0, 1, 1, 1, 0, 1, 0, 0, 1, 1, 1, 1, 0, 1, 1, 1, 1, 0, 1, 1, 1, 0,
       1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 1, 1, 0, 1, 0, 1, 1, 0, 1, 0,
       1, 1, 0, 1, 1, 0, 0, 1, 1, 0, 0, 1, 0, 1, 1, 1, 1, 1, 0, 0, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 1, 0, 1, 0, 0, 1, 1, 1, 0, 1, 1, 1,
       1, 1, 0, 0, 0, 1, 1, 1, 1, 0, 1, 0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 0,
       1, 0, 0, 1, 1, 0, 0, 1, 1, 1, 1, 1, 1, 0, 0,

- 0 ==> Not spam (Ham)
- 1 ==> Spam

Sample Prediction

As we need to provide the data or say text to the model in vectorize form (as we did) so for the future data or prediction, we should convert that data first into vectorize form.

In [63]:
predict_comment = ["Click on this link"]
vectorizer_predict = vectorizer.transform(predict_comment).toarray()
clf.predict(vectorizer_predict)

array([1], dtype=int64)

In [64]:
# predict_comment_2 = ["Awesome song love you!"]
# vectorizer_predict_2 = vectorizer.transform(predict_comment_2).toarray()
# clf.predict(vectorizer_predict_2)

In [65]:
if clf.predict(vectorizer_predict) == 1:
    print('Spam')
elif clf.predict(vectorizer_predict) == 0:
    print('Not Spam/ Ham')

Spam


### Saving Model

In [66]:
import pickle

In [67]:
NBC_Model = open("NBC_Model.pkl", "wb")

In [68]:
pickle.dump(clf, NBC_Model)

In [69]:
NBC_Model.close()